src/test/java/com/company/heap/RunningMedianTest.java

```diff
@@ -0,0 +1,29 @@
+package com.company.heap;
+
+import com.company.bitmanipulation.UnsetBit;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RunningMedianTest {
+    @ParameterizedTest
+    @MethodSource("testDataProvider")
+    public void testFindRunningMedian(int[] arr, int[] expectedResult) {
+        int[] result = new RunningMedian().findRunningMedian(arr);
+        assertEquals(Arrays.toString(expectedResult), Arrays.toString(result));
```

Comment:
You can use `assertArrayEquals` for asserting arrays

Reply:
changed